docs/app/pages/components/components-sections/input-controls/input-dropdown/input-dropdown.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, OnInit, Pipe, PipeTransform } from '@angular/core';
+import { BaseDocumentationSection } from '../../../../../components/base-documentation-section/base-documentation-section';
+import { DocumentationSectionComponent } from '../../../../../decorators/documentation-section-component';
+import { IPlayground } from '../../../../../interfaces/IPlayground';
+import { IPlaygroundProvider } from '../../../../../interfaces/IPlaygroundProvider';
+
+
+@Pipe({name: 'highlightSearch'})
+export class HighlightSearch implements PipeTransform {
+    transform(text: string, filter: string): string {
+        const highlightIndex = text.toLowerCase().indexOf(filter.toLowerCase());
+        return (highlightIndex < 0) ?
+               text :
+               text.substr(0, highlightIndex) +
+                 '<b>' + text.substr(highlightIndex, filter.length) + '</b>' +
+                 text.substr(highlightIndex + filter.length);
+    }
+}
+
+
+@Component({
+    selector: 'uxd-components-input-dropdown',
+    templateUrl: './input-dropdown.component.html',
+    styleUrls: ['./input-dropdown.component.less']
+})
+@DocumentationSectionComponent('ComponentsInputDropdownComponent')
+export class ComponentsInputDropdownComponent extends BaseDocumentationSection implements IPlaygroundProvider, OnInit {
+
+    selected: RadioOption;
+    optionList: ReadonlyArray<RadioOption> = [
+        { name: 'One' }, { name: 'Two' }, { name: 'Three' }, { name: 'Four' }
+    ];
+    filteredOptionList: ReadonlyArray<RadioOption> = this.optionList;
+    filter: string = '';
+    allowNull: boolean = false;
+    componentMaxHeight: number = 400;
+    placeholder: string = 'Type to search...';
+
+    playground: IPlayground = {
+        files: {
+            'app.component.html': this.snippets.raw.appHtml,
+            'app.component.ts': this.snippets.raw.appTs,
+            'app.component.css': this.snippets.raw.appCss
+        },
+        modules: [
+            {
+                imports: ['HighlightSearch'],
+                library: './app.component',
+                declaration: true
+            }, 
+            {
+                imports: ['AccordionModule', 'CheckboxModule', 'NumberPickerModule', 'RadioButtonModule', 'InputDropdownModule'],
+                library: '@ux-aspects/ux-aspects'
+            }
+        ]
+    };
+
+    constructor() {
+        super(require.context('./snippets/', false, /\.(html|css|js|ts)$/));
+    }
+
+    ngOnInit(): void {
+        this.filterList();
```

Comment:
Is this needed? The list already gets set with the initial items above?

Reply:
On the demo page, it is not needed, but in case you passed a non-empty filter into the component. I am not sure if this hint for the consumers of the component is necessary.

What do you think?

Reply:
I think it's probably best if we only include the code that is needed for the example to run. It's probably best to keep the example as simple as possible, as previously mentioned, people tend to just copy and paste code as is, so its probably best not to include code that isn't directly required and let them enhance the sample code as they need.

Reply:
OK, I removed it.